lib/showoff.rb

```diff
@@ -478,6 +479,178 @@ def process_content_for_replacements(content)
       result
     end
 
+    # This method defines several custom additions to markdown
+    # :matchers is an array of regular expressions that define the custom syntax
+    # :block defines a code block to be run against the match_data
+    # These are processed by gsub so the block should return the final result
+    def custom_syntax_templates
+      { 
+        "current" => {
+          :matchers => [
+            /~~~CURRENT_SLIDE~~~/
+          ],
+          :block => proc { |match_data|
+            @slide_count.to_s
+          }
+        },
+        "section_major" => {
+          :matchers => [
+            /~~~SECTION:MAJOR~~~/
+          ],
+          :block => proc { |match_data|
+            @section_major.to_s
+          }
+        },
+        "section_minor" => {
+          :matchers => [
+            /~~~SECTION:MINOR~~~/
+          ],
+          :block => proc { |match_data|
+            @section_minor += 1
+            @section_minor.to_s
+          }
+        },
+        "font_awesome" => {
+          :matchers => [
+            /\[(?<icon>fa-.*)\]/
+          ],
+          :block => proc { |match_data|
+            %Q(\n<i class="fa #{match_data[:icon]}"></i>\n)
+          }
+        },
+        "image" => {
+          :matchers => [
+            /\s+\!\[(?<classes>\.\S*)\s*(?<alt>.*?)\]\((?<src>[^\]]*)\)/
+          ],
+          :block => proc { |match_data|
+            classes = match_data[:classes].gsub('.', ' ')
+            %Q(\n<img src="#{match_data[:src]}" class="#{classes}" alt="#{match_data[:alt]}">\n)
+          }
+        },
+        "callout" => {
+          :matchers => [
+            /^\.(?=callout)(?<classes>.*)\s+(?<inner>.*?)$/
```

Comment:
Actually, what do you think of this:
 /^\.(?=callout)(?<classes>[\S]*?)\s(?<inner>.*?)\n\n/m

That would allow multiline callouts but break on two newlines.